app/javascript/components/TrackList.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { Track } from './TrackList/Track'
+import { TextFilter } from './TextFilter'
+import { useList } from '../hooks/use_list'
+import { useRequestQuery } from '../hooks/request_query'
+
+export function TrackList(props) {
+  const [request, setFilter, setSort, setPage] = useList(props.request)
+  const { status, data, isFetching } = useRequestQuery('track-list', request)
+
+  return (
+    <div className="track-list">
+      <TextFilter
+        filter={request.query.filter}
+        setFilter={setFilter}
+        id="track-list-filter"
+        placeholder="Search language tracks"
+      />
+      {status === 'error' && <p>Something went wrong</p>}
+      {isFetching && <p>Loading</p>}
```

Comment:
We now have several places where we have this loading element. Maybe we should extract it to a separate component?